src/site/content/en/blog/compute-pressure/index.md

```diff
@@ -0,0 +1,284 @@
+the hardware inside of a device is being pushed to its limit, and the only
+options are dissipating the associated heat, or deliberately slowing down the
+speed of the processors so that they'll run cooler. Most installed applications
+are able to detect when a device is under this type of CPU stress, and can
+selectively disable features or reduce their workloads in response.
+Increasingly, there are web applications that would benefit from the same
+information.
+
+[Compute Pressure](https://github.com/oyiptong/compute-pressure/) is a new,
+experimental API intended to allow web applications to gain insight into the CPU
+utilization of the browser's device. The API is designed as a callback mechanism
+that receives periodic updates, allowing developers to observe changes in CPU
+utilization over time. A web app that uses this API can make informed decisions
+about enabling or disabling heavyweight features in response to these
+observations. It's a feedback mechanism that unlocks new avenues for adaptive
+behavior, which can in turn lead to smoother, cooler experiences across a range
+of hardware devices.
+
+## Use cases
+
+The motivating use cases are helping video conferencing and video game web
+applications adapt to CPU utilization. The initial experimental implementation
+of the API aims to support the following potential decisions:
+
+### Video conferencing
+
+- Dynamically adjust the number of live video feeds displayed on the screen at
+  once.
+- Adjust the video resolution or frame rate.
+- Skip non-essential video processing, such as cosmetic filters.
+- Disable non-essential audio processing, such as noise suppression.
+- Skew the quality-vs-speed and size-vs-speed knobs towards "speed" in video and
+  audio encoding.
+
+### Video games
+
+- Use lower-quality video and audio assets as needed.
+- Disable optional effects like reflections, glare, or particle animations that
+  don't affect gameplay.
+- Tweak quality-vs-speed knobs in the game's rendering engine, affecting areas
+  like shadows quality, texture filtering, and view distance.
+
+The secondary use case is measuring the CPU resource consumption of a feature.
+This ultimately supports the main goal of avoiding driving user devices into
+high CPU utilization. The API aims to help developers make the following types
+of decisions:
+
+- Compare the CPU consumption of alternative implementations of the same
+  feature, to find out which is more efficient. The API supports measuring CPU
+  utilization in the field via
+  [A/B tests](https://en.wikipedia.org/wiki/A/B_testing), because a feature's
+  CPU utilization depends on the hardware it's running on. Most developers
+  cannot afford performance measurement labs covering all the devices owned by
+  their users.
+- Estimate the impact of enabling a feature on CPU consumption. This cost
+  estimate feeds into the decisions outlined in the primary use case.
+
+## Core concepts
```

Comment:
Currently, the TOC on web.dev doesn't show different levels. There's also an implied commitment in any article that core concepts are covered. Therefore I recommend removing this heading and promoting the next two.

Reply:
Okay.